spec/controllers/media_controller_spec.rb

```diff
@@ -115,4 +121,42 @@
       end
     end
   end
+
+  describe '#verify_token' do
+    let(:id) { 'ab123cd4567' }
+    let(:file_name) { 'interesting_video.mp4' }
+    let(:ip_addr) { '192.168.1.100' }
+    let(:token) { StacksMediaToken.new(id, file_name, ip_addr) }
+    let(:encrypted_token) { token.to_encrypted_string }
+
+    it 'verifies a valid token' do
+      get :verify_token, token_string: encrypted_token, id: id, file_name: file_name, user_ip_addr: ip_addr
+      expect(response.body).to eq 'valid token'
+      expect(response.status).to eq 200
+    end
+
+    it 'rejects a token with a corrupted encrypted token string' do
+      get :verify_token, token_string: "#{encrypted_token}aaaa", id: id, file_name: file_name, user_ip_addr: ip_addr
+      expect(response.body).to eq 'invalid token'
+      expect(response.status).to eq 403
+    end
+
+    it 'rejects a token for the wrong id' do
+      get :verify_token, token_string: encrypted_token, id: 'zy098xv7654', file_name: file_name, user_ip_addr: ip_addr
+      expect(response.body).to eq 'invalid token'
+      expect(response.status).to eq 403
+    end
+
+    it 'rejects a token for the wrong file name' do
+      get :verify_token, token_string: encrypted_token, id: id, file_name: 'some_other_file.mp3', user_ip_addr: ip_addr
+      expect(response.body).to eq 'invalid token'
+      expect(response.status).to eq 403
+    end
+
+    it 'rejects a token from the wrong IP address' do
+      get :verify_token, token_string: encrypted_token, id: id, file_name: file_name, user_ip_addr: '192.168.1.101'
+      expect(response.body).to eq 'invalid token'
+      expect(response.status).to eq 403
+    end
```

Comment:
yeah, actually, the better way to do this would probably be to test that `token_still_valid?` calls `StacksMediaToken.verify_encrypted_token_string` with the expected params, and that `verify_token` does the right thing with the result of `token_still_valid?` (by mocking the `token_still_valid?` call).  the specific validity conditions are tested in `stacks_media_token_spec.rb`, which seems correct/sufficient (since that's where the validity checking logic lives).